### PR TITLE
Documentation: bring in line to 0.3.0

### PR DESCRIPTION
**Pull request by hsanjuan**
Review documentation to be in line with latest updates to Raft and
any other feature introduced since 0.12.0.

License: MIT
Signed-off-by: Hector Sanjuan <hector@protocol.ai>

### CODE REVIEW

**Comment by hsanjuan** [comment]
This is a doc-only PR. Yes, we haven't released 0.3.0, but this is in line with current master.

File: docs/ipfs-cluster-guide.md [hunk 1]
```diff
@@ -248,16 +271,20 @@ The reason pins (and unpin) requests are queued is because ipfs only performs on
...
+## Cluster monitoring and pin failover
 
 ipfs-cluster includes a basic monitoring component which gathers metrics and triggers alerts when a metric is no longer renewed. There are currently two types of metrics:
 
 * `informer` metrics are used to decide on allocations when a pin request arrives. Different "informers" can be configured. The default is the disk informer using the `freespace` metric.
-* a `ping` metric is used to monitor peers.
+* a `ping` metric is used to signal that a peer is alive.
 
-Every metric carries a Time-To-Live associated with it. ipfs-cluster peers pushes metrics from every peer to the cluster Leader in TTL/2 intervals. When a metric for an existing cluster peer stops arriving and previous metrics have outlived their Time-To-Live, the monitoring component triggers an alert for that metric.
+Every metric carries a Time-To-Live associated with it. This TTL can be configued in the `informers` configuration section. The `ping` metric TTL is determined by the `cluster.monitoring_ping_interval`, and equals to 2x its value.
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: Typo: "is equal to 2x" or "equals 2x"

File: docs/ipfs-cluster-guide.md [hunk 2]
```diff
@@ -248,16 +271,20 @@ The reason pins (and unpin) requests are queued is because ipfs only performs on
...
 
-ipfs-cluster will react to `ping` metrics alerts by searching for pins allocated to the alerting peer and triggering re-pinning requests for them. If the node is down, informer metrics are likely to be also invalid, causing the content to be given allocations from different peers.
+Every ipfs-cluster peers push metrics to the cluster Leader regularly. This happens TTL/2 intervals for the `informer` metrics and in `cluster.monitoring_ping_interval` for the `ping` metrics.
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: Typo: "peers push metrics" -> "peer pushes metrics"

Comment (ZenGround0) [reply docs/ipfs-cluster-guide.md]: Typo "This happens TTL/2 intervals for ..." missing an "in"

File: docs/ipfs-cluster-guide.md [hunk 3]
```diff
@@ -291,18 +318,17 @@ This means that the top cluster will think that it is performing requests to an
 
 This allows to scale ipfs-cluster deployments and provides a method for building ipfs-cluster topologies that may be better adapted to certain needs.
 
-Note that **this feature has not been extensively tested**.
+Note that **this feature has not been extensively tested**, but we aim to introduce improvements and fully support it in the mid-term.
 
 
 ## Security
 
 ipfs-cluster peers communicate which eachother using libp2p-encrypted streams (`secio`), with the ipfs daemon using plain http, provide an HTTP API themselves (used by `ipfs-cluster-ctl`) and an IPFS Proxy. This means that there are four endpoints to be wary about when thinking of security:
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: Typo: "communicate which eacchother"

File: docs/ipfs-cluster-guide.md [hunk 4]
```diff
@@ -320,7 +348,7 @@ When the shared state format has changed, a state migration will be required. ip
 
 Open an issue on [ipfs-cluster](https://github.com/ipfs/ipfs-cluster) or ask on [discuss.ipfs.io](https://discuss.ipfs.io).
 
-If you want to collaborate in ipfs-cluster, open an issue and ask about what you can do.
+If you want to collaborate in ipfs-cluster, look at the list of open issues. The are many are conveniently marked with [HELP WANTED](https://github.com/ipfs/ipfs-cluster/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) and organized by difficulty. If in doubt, just ask!
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: Typo: "The are many are "

File: docs/ipfs-cluster-guide.md [hunk 5]
```diff
@@ -94,57 +99,68 @@ Each section of the configuration file and the options in it depend on their ass
...
+When filling in `peers` with some other peers' listening multiaddresses (i.e. `/ip4/192.168.1.103/tcp/9096/ipfs/QmQHKLBXfS7hf8o2acj7FGADoJDLat3UazucbHrgxqisim`), the peer will expect to be part of a cluster with the given `peers`. On boot, it will wait to learn who is the leader (see `raft.wait_for_leader_timeout` option) and sync it's internal state up to the last known state before becoming `ready`.
+
+If you are using the `peers` configuration value, then **it is very important that the `peers` configuration value in all cluster members is the same for all peers,** that is, contain the multiaddresses for the other peers in the cluster. It may contain this peer's own multiaddress too (but it will removed on the next shutdown). If `peers` is not correct for all peer members, your node might not start or misbehave in not obvious ways.
+
+You are expected to start nodes at the same time when using this method. If half of them are not started, they will fail to elect a cluster leader. If there are peers missing, the cluster will not be in a healthy state (error messages will be displayed). The cluster will operate, as long as a majority of peers is up.
```
[the remark this answers is not in the document]
Comment (hsanjuan) [reply docs/ipfs-cluster-guide.md]: added that the delay is controlled by `wait_for_leader_timeout`

**Comment by hsanjuan** [comment]
Feedback addressed. I think I will also add the changes from #226 to this PR once that's merged.

Thanks!

**Comment by hsanjuan** [comment]
@ZenGround0 added one last commit. We can merge after your OK. Thanks!

File: architecture.md [hunk 6]
```diff
@@ -125,21 +124,13 @@ Notes:
 * If it's via an API request, it involves an RPC request to the cluster main component.
 * `Cluster.PeerAdd()`
 * Figure out the real multiaddress for that peer (the one we see).
-* Let the `PeerManager` component know about the new peer. This adds it to the Libp2p host and notifies any component which needs to know
-about peers. This means also ability to perform RPC requests to that peer.
+* Broadcast the address to every cluster peer and add it to the host's libp2p peerstore. This means also ability to perform RPC requests to that peer.
```
Comment (ZenGround0) [inline architecture.md]: Typo "This means also ability to perform RPC requests to that peer" is grammatically off.  Perhaps "This gives each member of the cluster the ability to perform RPC  requests..."?

File: docs/ipfs-cluster-guide.md [hunk 7]
```diff
@@ -146,18 +146,18 @@ Each section of the configuration file and the options in it depend on their ass
...
 
-If you are using the `peers` configuration value, then **it is very important that the `peers` configuration value in all cluster members is the same for all peers,** that is, contain the multiaddresses for the other peers in the cluster. It may contain this peer's own multiaddress too (but it will removed on the next shutdown). If `peers` is not correct for all peer members, your node might not start or misbehave in not obvious ways.
+If you are using the `peers` configuration value, then **it is very important that the `peers` configuration value in all cluster members is the same for all peers**: it should contain the multiaddresses for the other peers in the cluster. It may contain a peer's own multiaddress too (but it will removed automatically). If `peers` is not correct for all peer members, your node might not start or misbehave in not obvious ways.
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: Typo: but it will removed automatically

File: docs/ipfs-cluster-guide.md [hunk 8]
```diff
@@ -146,18 +146,18 @@ Each section of the configuration file and the options in it depend on their ass
...
+* The *default* case - peer has not been removed and `cluster.leave_on_shutdown` is `false`: in this case the peer has not left the consensus peerset, and you may start the peer again normally. Do not manually update `cluster.peers`, even if other peers have been removed from the cluster.
+* The *left the cluster* case - peer has been manually removed or `cluster.leave_on_shutdown` is `true`: in this case, unless the peer died, it has probably been removed from the consensus (you can check if its missing from `ipfs-cluster-ctl peers ls` on a running peer). This will mean that the state of the peer has been cleaned up (see the Dynamic Cluster Membership considerations below), and the last known `cluster.peers` have been moved to `cluster.bootstrap`. When the peer is restarted, it will attempt to rejoin the cluster from which it was removed by using the addresses in `cluster.bootstrap`.
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: Subtle typo: (you can check if its missing from...) -> (you can check if it's missing from ...)

File: docs/ipfs-cluster-guide.md [hunk 9]
```diff
@@ -174,11 +174,11 @@ In order to work, Raft elects a cluster "Leader", which is the only peer allowed
...
 
-The "peer add" and "peer remove" operations also trigger log entries and fully depend on a healthy consensus status. Modifying the cluster peers is a tricky operation because it requires informing every peer of the new peer set. If a peer is down during this operation, it is likely that it will not learn about it when it comes up again. Thus, it is recommended to bootstrap the peer again when the cluster has changed in the meantime.
+The "peer add" and "peer remove" operations also trigger log entries (internal to Raft) and depend too on a healthy consensus status. Modifying the cluster peers is a tricky operation because it requires informing every peer of the new peer's multiaddresses. If a peer is down during this operation, the operation will fail, as otherwise that peer will not know how to contact the new member. Thus, it is recommended delete and bootstrap any peer that is offline before making changes to the peerset.
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: When you say "delete and bootstrap" do you mean "remove and bootstrap"?

File: docs/ipfs-cluster-guide.md [hunk 10]
```diff
@@ -174,11 +174,11 @@ In order to work, Raft elects a cluster "Leader", which is the only peer allowed
...
+By default, the consensus log data is backed in the `ipfs-cluster-data` subfolder, next to the main configuration file. This folder stores two types of information: the **boltDB** database storing the Raft log, and the state snapshots. Snapshots from the log are performed regularly when the log grows too big (see the `raft` configuration section for options). When a peer is far behind in catching up with the log, Raft may opt to send a snapshot directly, rather than to send every log entry that make up the state individually. This data is initialized on the first start of a cluster peer and maintained throughout its life. Removing or renaming the `ipfs-cluster-data` folder effectively resets the peer to a clean state. Only peers with a clean state should bootstrap to already running clusters.
```
Comment (ZenGround0) [inline docs/ipfs-cluster-guide.md]: Subtle typo: "every log entry that make up" should be either "every log entry that makes up" or "all log entries that make up"

**Comment by hsanjuan** [comment]
all fixed. It's amazing how many typos I make. I should pay more attention. Thanks!

**Comment by ZenGround0** [comment]
Surely fewer typos than if I were to write this in Spanish.  Thanks for the docs, the new behavior is very clear.

**Pull request merged**